packages/kbn-check-mappings-update-cli/src/extract_mappings_from_plugins.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import ChildProcess from 'child_process';
+import { Readable } from 'stream';
+
+import * as Rx from 'rxjs';
+
+import { REPO_ROOT } from '@kbn/repo-info';
+import { SomeDevLog } from '@kbn/some-dev-log';
+import { observeLines } from '@kbn/stdio-dev-helpers';
+import type { SavedObjectsTypeMappingDefinitions } from '@kbn/core-saved-objects-base-server-internal';
+
+import type { Result } from './extract_mappings_from_plugins_worker';
+
+function routeToLog(readable: Readable, log: SomeDevLog, level: 'debug' | 'error') {
+  return observeLines(readable).pipe(
+    Rx.tap((line) => {
+      log[level](line);
+    }),
+    Rx.ignoreElements()
+  );
+}
+
+/**
+ * Run a worker process that starts the core with all plugins enabled and sends back the
+ * saved object mappings for all plugins. We run this in a child process so that we can
+ * harvest logs and feed them into the logger when debugging.
+ */
+export async function extractMappingsFromPlugins(
+  log: SomeDevLog
+): Promise<SavedObjectsTypeMappingDefinitions> {
+  log.info('Loading core with all plugins enabled so that we can get all savedObject mappings...');
+
+  const fork = ChildProcess.fork(require.resolve('./extract_mappings_from_plugins_worker.ts'), {
```

Comment:
FMI: what's the value of forking here instead of just running the kbn server in the current process?

Reply:
This was done my me and @spenceralger primarily to control console output better. I agree that there probably isn't benefit from a processing perspective.